packages/flutter/lib/src/widgets/form.dart

```diff
@@ -430,8 +434,8 @@ class FormFieldState<T> extends State<FormField<T>> {
 
   @override
   Widget build(BuildContext context) {
-    // Only autovalidate if the widget is also enabled
-    if (widget.autovalidate && widget.enabled)
+    // Only autovalidate if the widget is also enabled and if its content has changed
+    if (widget.autovalidate && widget.enabled && _valueChanged)
```

Comment:
what if validator has changed?

Reply:
instead of using the _valueChanged you can cache the last validated value and only validate when the value is different.

Reply:
But if I cache the last value (lets say the first value that is a empty string) then we will not be able to validate if the value is a empty string.

Reply:
Working on this

Reply:
I see, so the use case is when form first build we don't want to validate until user touches it. but then once the user touches it, we want to start validate even if the form is still empty after user touched it.

Reply:
I think you also need to somehow set _valueChanged back to false after validation in this current implementation.

Reply:
Why we need to say "value was not changed" after each validation?

Reply:
Sorry I was confused a bit at the name valueChanged, I thought it meant the value has changed since last validation. it should be something like stateIsDirty or startsValidation. There is another PR tries to achieve the same thing https://github.com/flutter/flutter/pull/56132.

In this case the validator change does not really matter here.

Reply:
Right